packages/tools/viewer-alpha/src/viewer.ts

```diff
@@ -0,0 +1,240 @@
+        options?: ViewerOptions
+    ) {
+        const finalOptions = { ...defaultViewerOptions, ...options };
+        this._scene = new Scene(this._engine);
+        this._scene.clearColor = finalOptions.backgroundColor;
+        this._camera = new ArcRotateCamera("camera1", 0, 0, 1, Vector3.Zero(), this._scene);
+        this._camera.attachControl();
+        this._reframeCamera(); // set default camera values
+
+        // TODO: render at least back ground. Maybe we can only run renderloop when a mesh is loaded. What to render until then?
+        this._engine.runRenderLoop(() => {
+            this._scene.render();
+        });
+
+        options?.onInitialized?.({ scene: this._scene, camera: this._camera });
+    }
+
+    /**
+     * Loads a 3D model from the specified URL.
+     * @remarks
+     * If a model is already loaded, it will be unloaded before loading the new model.
+     * @param url The URL of the model to load.
+     * @param abortSignal An optional signal that can be used to abort the loading process.
+     */
+    public async loadModelAsync(url: string, abortSignal?: AbortSignal): Promise<void> {
+        if (this._isDisposed) {
+            throw new Error("Viewer is disposed");
+        }
+
+        this._loadModelAbortController?.abort();
+        const abortController = (this._loadModelAbortController = new AbortController());
+
+        const throwIfAborted = () => {
+            // External cancellation
+            abortSignal?.throwIfAborted();
+
+            // Internal cancellation
+            abortController.signal.throwIfAborted();
+        };
+
+        await this._loadModelLock.lockAsync(async () => {
+            throwIfAborted();
+            this._assetContainer?.dispose();
+            this._assetContainer = await SceneLoader.LoadAssetContainerAsync("", url, this._scene);
+            this._assetContainer.addAllToScene();
+            this._reframeCamera();
+        });
+    }
+
+    /**
+     * Loads an environment texture from the specified URL and sets up a corresponding skybox.
+     * @remarks
+     * If no URL is provided, a default hemispheric light will be created.
+     * If an environment is already loaded, it will be unloaded before loading the new environment.
+     * @param url The URL of the environment texture to load.
+     * @param abortSignal An optional signal that can be used to abort the loading process.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public async loadEnvironmentAsync(url: Nullable<string | undefined>, abortSignal?: AbortSignal): Promise<void> {
+        if (this._isDisposed) {
+            throw new Error("Viewer is disposed");
+        }
+
+        this._loadEnvironmentAbortController?.abort();
+        const abortController = (this._loadEnvironmentAbortController = new AbortController());
+
+        const throwIfAborted = () => {
+            // External cancellation
+            abortController.signal.throwIfAborted();

```

Comment:
Should this first one be abortSignal?.throwIfAborted()?

Reply:
Lol yes. Will fix.